### ISSUE TITLE
SSO on link unfurling show nothing

### ISSUE THREAD

**Issue by wenytang-ms** [OfficeDev/Microsoft-Teams-Samples#407]
Hi, your Teams App works fine on query message extension, but link unfurling not.
The Samples name is app-sso, and I tried the node.js one. 
https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-sso/nodejs
When I enter the link of the relevant domain name of botframework.com, if I have not logged in, the teams do not show anything as following
![image](https://user-images.githubusercontent.com/75360946/180733882-a1bf6dbb-573a-4ff5-9053-63b729b6d591.png)

I intercepted the information related to the operating environment at this time and provided it for your reference
![image](https://user-images.githubusercontent.com/75360946/180734201-573c047b-b560-4113-9a33-3152abbc73c2.png)
![image](https://user-images.githubusercontent.com/75360946/180734261-c1d19f56-b6ae-4f47-87ff-c869abdfbf9f.png)
request
![image](https://user-images.githubusercontent.com/75360946/180734464-cf182b18-e273-4d62-8690-944a8dad942b.png)
and response
![image](https://user-images.githubusercontent.com/75360946/180734542-02653fe7-cdf4-419d-b7e9-494581dfe16f.png)

**Comment by Meghana-MSFT** [comment]
we are looking into this issue, we will get back to you.

**Comment by wenytang-ms** [comment]
Or please describe this situation and what the expected performance should be like

**Comment by Meghana-MSFT** [comment]
@wenytang-ms - We tried the above sample with C#. As you can see in the attached screenshot the link got unfurled even without login. Also please note that we need to send a message to the bot to get the login option. Please send a message to get the sign in option and to start interacting with the bot.

For link unfurling please check these samples - 

1. https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/msgext-link-unfurling-meeting/csharp
2. https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/msgext-link-unfurling-reddit/csharp

![MicrosoftTeams-image (37)](https://user-images.githubusercontent.com/81138249/181266058-ef576787-58d5-4a29-881f-80f93d2184b3.png)

**Comment by wenytang-ms** [comment]
Thanks for your reply.  Let me rephrase my question.
Once there only Message Extension without any bot command registered. What I want to know is the SSO process only in the link unfurling scenario.
this is my repro steps:
1. register an AAD App
2. create azure bot
3. config the bot OAuth connection
4. start the node.js server and client
5. enter any botframework.com domain url link, will trigger the SSO process, such as https://profile.botframework.com
6. there no reaction or response on teams, no login prompt or consent requirement 

once enter the https://profile.botframework.com URL, the message extension needs to get SSO token to exchange the Graph token with profile scope, then Graph Client will get the profile image and return the ActionCard.
What's the SSO prompt or consent requirement should be in link unfurling?

**Comment by wenytang-ms** [comment]
Just like query in message extension, the extension will ask me to login/consent as following:
![image](https://user-images.githubusercontent.com/75360946/181409145-a0b76e0e-7725-42b7-902d-27ba0136d104.png)

**Comment by wenytang-ms** [comment]
is there any response?

**Comment by sayali-MSFT** [comment]
Currently We don’t have ETA or Update on this thread. We will update you soon.

**Comment by Meghana-MSFT** [comment]
@wenytang-ms - Apologies for the delay in response. Currently authentication is supported in Bots, Tabs and message extensions only. Please see this document related to the same - https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/authentication/authentication.
That is why we see a login/signin message first when you search for anything in the message extension. And for link unfurling scenario it just unfurls the link and does not ask for any login consent.

**Comment by wenytang-ms** [comment]
@Meghana-MSFT thanks for your response. 
I try the sample on link unfurling with response type as auth instead of silentAuth, if user not login, the link unfurling will ask user to login. I tried this sample with bot config set as null in manifest.json.
![image](https://user-images.githubusercontent.com/75360946/185873146-bce2e30c-78ee-46e7-bda1-8aac3b792ff3.png)

So I just wondering what's the link unfurling SSO login should be?

**Comment by wenytang-ms** [comment]
@Meghana-MSFT Thanks for your response.
As you say that unfurls the link and does not ask for any login consent.
I tried this sample with link unfurling, at meanwhile, set the 'handleTeamsAppBasedLinkQuery' return auth type instead of silentAuth, and I got this response without SSO, but there has login button.
![image](https://user-images.githubusercontent.com/75360946/185874356-0016aedd-e3f2-4e9d-8313-702a387ddef8.png)
I also set bot config as null in manifest.json file. zip it and upload to teams.
So I think login in link unfurling has response, and I'm wondering if there has SSO consent login response?

**Comment by Meghana-MSFT** [comment]
We tried [app-sso](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-sso/nodejs) sample again and we could see the login button when we are not logged in. We have not changed the return type to auth as well.
![signin](https://user-images.githubusercontent.com/81138249/186649756-f0cf8015-7500-4433-85d2-abd75abeaefd.PNG)

Logged in using the signin button 
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/81138249/186650099-4adcab00-95a8-43dd-802b-dfdf4192a590.png)

https://profile.botframework.com got unfurled and displayed the profile card.
![image](https://user-images.githubusercontent.com/81138249/186650129-dc1d8623-d74a-4d0a-b08e-cc80b51e5927.png)

**Comment by wenytang-ms** [comment]
I fetch the latest code sample and try both node.js and csharp. Node.js fail, but csharp success. I know you tried the csharp one.
But your node.js sample is different with csharp one. 

Node.js response to handleTeamsMessagingExtensionQuery with silentAuth, which is [enable SSO](https://docs.microsoft.com/en-us/microsoftteams/platform/messaging-extensions/how-to/enable-sso-auth-me)
but csharp not! 
what csharp do just the login, and this sample cannot follow [enable the sso step](https://docs.microsoft.com/en-us/microsoftteams/platform/messaging-extensions/how-to/enable-sso-auth-me#enable-sso-for-message-extensions-and-link-unfurling)

So, does link unfurling really support SSO?
![image](https://user-images.githubusercontent.com/75360946/187381854-bb9ee4ff-9269-41cf-aca0-633a19f98258.png)
if not, the doc is wrong, and the sample in this doc cannot work, do you have PM? can he or she explain?

**Comment by wenytang-ms** [comment]
It's been a month and you still don't know what my question is!

**Comment by wenytang-ms** [comment]
any response?

**Comment by Meghana-MSFT** [comment]
Apologies for the delay in response, Link unfurling supports SSO. And as you rightly pointed out it should be silentAuth instead of auth. We are updating the code.

**Comment by wenytang-ms** [comment]
The code doesn't seem to be updated.
Did you also encounter the same problem in SSO as me with sample code?

**Comment by Meghana-MSFT** [comment]
Yes, and we are internally working to update the code. We will let you know once the sample is updated.
